### PR TITLE
handle bad MATCH_ENV

### PR DESCRIPTION
**Pull request by orimwix**
### 🔦 Summary
background:
In Yoshi 4, there is no longer an env called `component`, yet projects migrating from Yoshi 3 might not know this, and set `MATCH_ENV=component`. 
If they do, jest is run with no configuration at all, and get a long and meaningless list of errors.

solution:
1) If no env is matched
- configure Jest to not run any test.
- Print a meaningful error to the console
2) Add a note to the migration docs

### CODE REVIEW

File: packages/jest-yoshi-preset/jest-preset.js [hunk 1]
```diff
@@ -100,7 +108,6 @@ module.exports = {
             '^.+\\.jsx?$': require.resolve('./transforms/babel'),
             '^.+\\.tsx?$': require.resolve('ts-jest'),
             '\\.st.css?$': require.resolve('@stylable/jest'),
-            '\\.(gql|graphql)$': require.resolve('jest-transform-graphql'),
```
Comment (yurynix) [inline packages/jest-yoshi-preset/jest-preset.js]: Why is that required?

File: packages/jest-yoshi-preset/jest-preset.js [hunk 2]
```diff
@@ -115,6 +122,12 @@ module.exports = {
           setupFilesAfterEnv,
         };
       }),
+    // fallback if no env was matched
```
Comment (yurynix) [inline packages/jest-yoshi-preset/jest-preset.js]: Why we need two of those?, looks like a duplicate of the one below:
```
    // workaround for https://github.com/facebook/jest/issues/5866
    {
      displayName: 'dummy',
      testMatch: ['dummy'],
      modulePathIgnorePatterns,
    },
```

**Comment by yurynix** [comment]
Thank you for taking the time to create a PR for that!
I had a few questions that I hope you could answer.

File: packages/jest-yoshi-preset/jest-preset.js (hunk 1 above)
Comment (yurynix) [reply packages/jest-yoshi-preset/jest-preset.js]: That also fails the `test:app-flow` tests.

Comment (orimwix) [reply packages/jest-yoshi-preset/jest-preset.js]: looks like a keyboard accident, oops

File: packages/jest-yoshi-preset/jest-preset.js (hunk 2 above)
Comment (orimwix) [reply packages/jest-yoshi-preset/jest-preset.js]: the original workaround was needed because of an issue in Jest that ignores the config if there is only one project.
the original workaround fixed a case where the user chose 1 project, and artificially increased the number to 2. Now I suggest a workaround for a case where the user chose 0 valid projects, therefore need to add 2 fake projects to satisfy Jest

**Comment by orimwix** [comment]
Hi, any reason not to merge this?

File: docs/migration/version-4.md [hunk 3]
```diff
@@ -141,6 +141,8 @@ To make migration easier, please start by running a helper script, which will ch
 curl https://gist.githubusercontent.com/ronami/1608dc49efc166bb6e15a21f7073cb79/raw | node
 ```
 
+note: `MATCH_ENV=component` should be replaced with `MATCH_ENV=spec`
```
Comment (ranyitz) [inline docs/migration/version-4.md]:
```suggestion
__Note:__ If you're using `MATCH_ENV=component` it should be replaced with `MATCH_ENV=spec`
```
